src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16450,11 +16452,74 @@ def str_len(self, **kwargs: Any) -> "SnowflakeQueryCompiler":
                 )
             )
 
-    def str_ljust(self, width: int, fillchar: str = " ") -> None:
-        ErrorMessage.method_not_implemented_error("ljust", "Series.str")
+    def str_ljust(self, width: int, fillchar: str = " ") -> "SnowflakeQueryCompiler":
+        """
+        Pad right side of strings in the Series/Index.
+
+        Equivalent to str.ljust().
+
+        Parameters
+        ----------
+        width : int
+            Minimum width of resulting string; additional characters will be filled with fillchar.
+        fillchar : str
+            Additional character for filling, default is whitespace.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler representing result of the string operation.
+        """
+        if not isinstance(width, int):
+            raise TypeError(
+                f"width must be of integer type, not {type(self._modin_frame.data_column_snowflake_quoted_identifiers[0]).__name__}"
+            )
+
+        def output_col(column: SnowparkColumn) -> SnowparkColumn:
+            new_col = rpad(
+                column,
+                greatest(length(column), pandas_lit(width)),
+                pandas_lit(fillchar),
+            )
+            return self._replace_non_str(column, new_col)
```

Comment:
this will coerce non-string columns to string, right? is this desired behavior? it seems like in native pandas it will error out

Reply:
I think this is pandas behavior too. When I remove this step, tests fail because output is not the same as native pandas. It's also used for other string functions.

Reply:
I got this from native pandas:
```
>>> native_pd.Series([1]).str.ljust(2)
AttributeError: Can only use .str accessor with string values!
```

If it turns out this extra line is necessary to properly handle VARIANT columns, then I think it's fine to leave in.